index.js

```diff
@@ -171,7 +171,10 @@ exports.parseLink = function parseBlob (ref) {
   var match = parseLinkRegex.exec(ref)
   if (match && match[1]) {
     if (match[3]) {
-      return {link: match[1], query: Querystring.parse(match[4])}
+      var query = Querystring.parse(match[4])
+      // unbox keys have a '+' in them that is parsed into a ' ', this changes it back
+      if (query.unbox) query.unbox = query.unbox.replace(' ', '+')
```

Comment:
For multiple replacements I think we'd need something like:

```js
if (query.unbox) query.unbox = query.unbox.replace(/ /g, '+')
```

There's another funky edge-case here where `unbox` could be a truthy non-string (like `42`), and `(42).replace()` throws a nasty error. Maybe something like this?

```js
if (isString(query.unbox)) query.unbox = query.unbox.replace(/ /g, '+')
```

Could be overkill.

Reply:
if unbox isn't a base64 for a 32 byte key, it's not a valid key, so it should be ignored in that case.